### ISSUE TITLE
specifying punctures like e^{2 pi i/3}

### ISSUE THREAD

**Issue by neitzke** [chan-y-park/loom#45]
I just tried to make a network (using the web UI) corresponding to the differentials

phi_2 = 0
phi_3 = 1 / (z^3-1)^2  dz^3

The punctures should be at the cube roots of unity; but I don't know how to specify this in the UI. I tried putting in `[1,E^(2 Pi I/3), E^(4 Pi I/3)]` but this didn't seem to work -- in the output window I just get

```
Start loom, uuid = a4e911c8-27b6-4743-9d56-fb402f3c0f65
18196: Started @ 2016-01-30 22:25:20
```

and nothing else. Is there a way to make this example work?

**Comment by plonghi** [comment]
Hi Andy,

to specify the punctures try using instead 
[1, E^(2 \* 3.14159 \* I/3), E^(4 \* 3.14159 \* I/3)]
because "Pi" is not yet interpreted (a minor fix that can be quickly implemented in the master branch, but won't be available on the web interface for a while).
Also note the use of the multiplication sign " \* " which is important for the parser.

One more thing, in phi_3 there should not be dz^3 when entered in the web UI.

Finally, with these specifications the example will correctly interpret the differentials. But then it will complain that there are no branch points for this theory, so no spectral network. 
Please let us know if you still have trouble.

**Comment by chan-y-park** [comment]
Hi Andy, 

Information regarding punctures is first saved as a string, and then interpreted by `SymPy`. So any correct `SymPy` expression will go through `loom`. For example, we can use
`[1,E^(2*pi*I/3), E^(4*pi*I/3)]`
for the punctures that you want to specify. Then as Pietro says loom doesn't find any branch point for the data you specified.

P.S. to Pietro: I guess Andy used dz^3 in his text for us and didn't use it when inputting the data into `loom`. :)

**Comment by neitzke** [comment]
Thanks, that clears it up!

The spectral network in this case actually should not be trivial -- a 
puncture with zero residue does emit some walls. You can see that by 
making a very small mass deformation, which will have the effect of 
putting two branch points very close to the puncture. But this is a new 
kind of special case which I guess hasn't yet made it into the code; and 
in fact I don't think I know what is the correct general rule to put in 
(it might be interesting to work it out!)

**Comment by chan-y-park** [comment]
Oh, I see, that's why `loom` didn't find any ramification point. I think I implemented a routine that deals with a massless regular puncture, which is exactly what we have when branch points collide with a regular puncture as mass parameters become zero, but it's lost in the middle of many revisions.

I think I know what the general rule is as far as massless regular punctures of A-type spectral networks are concerned, which passed a few nontrivial tests when I studied AD theories. Here is an excerpt from my note:
https://github.com/chan-y-park/loom/blob/master/documentation/s_walls_around_a_regular_puncture.pdf
